lib/chef_apply/action/converge_target.rb

```diff
@@ -115,13 +116,16 @@ def create_remote_config(dir)
       remote_config_path
     end
 
-    def create_remote_handler(dir)
-      remote_handler_path = File.join(dir, "reporter.rb")
+    def create_remote_handler(remote_dir)
+      remote_handler_path = File.join(remote_dir, "reporter.rb")
       begin
-        handler_file = Tempfile.new
+        # TODO - why don't we upload the original remote_handler_path instead of making a temp copy?
```

Comment:
This is good question. I don't remember.

Reply:
I'll leave the TODO and try it out somewhere along the way.